pandas/core/internals.py

```diff
@@ -1809,6 +1817,11 @@ def putmask(self, mask, new, align=True, inplace=False, axis=0,
         new_values = self.values if inplace else self.copy().values
         new_values, _, new, _ = self._try_coerce_args(new_values, new)
 
+        if is_sparse(new_values):
```

Comment:
similar to above

Reply:
I'll get used to it. :) Done. I hope that's what you meant. Much simpler now altogether.